src/main/kotlin/uk/gov/justice/digital/hmpps/educationemploymentapi/config/ExceptionAdvice.kt

```diff
@@ -112,19 +113,19 @@ class ControllerAdvice {
     log.info("Validation exception: ${e.message}", e)
     val genericMessage = "Unacceptable JSON " + e.message
     var errorDetails: String? = genericMessage
-    if (e.cause is MissingKotlinParameterException) {
-      val mkpx: MissingKotlinParameterException = e.cause as MissingKotlinParameterException
+    if (e.cause is MismatchedInputException) {
```

Comment:
replacing deprecated class - `MissingKotlinParameterException` extends `MismatchedInputException` so this should still work